homeassistant/components/lacrosse_view/config_flow.py

```diff
@@ -0,0 +1,123 @@
+"""Config flow for LaCrosse View integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from lacrosse_view import LaCrosse, Location, LoginError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("username"): str,
+        vol.Required("password"): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> list[Location]:
+    """Validate the user input allows us to connect."""
+
+    api = LaCrosse(async_get_clientsession(hass))
+
+    try:
+        if not await api.login(data["username"], data["password"]):
```

Comment:
So something like this would be good?
```suggestion
        await api.login(data["username"], data["password"])
        
        locations: list[Location] = await api.get_locations()
```